ring/spread_minimizing_token_generator_test.go

```diff
@@ -0,0 +1,553 @@
+package ring
+
+import (
+	"container/heap"
+	"fmt"
+	"math"
+	"math/rand"
+	"os"
+	"testing"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/exp/slices"
+)
+
+var (
+	zones             = []string{"zone-a", "zone-b", "zone-c"}
+	tokensPerInstance = 512
+)
+
+func TestSpreadMinimizingTokenGenerator_GetInstanceID(t *testing.T) {
+	tests := map[string]struct {
+		instanceID    string
+		expectedID    int
+		expectedError error
+	}{
+		"instance-zone-a-10 is correct": {
```

Comment:
I know it works, but can you add this test case that includes a dash in the instance name?
```
		"store-gateway-zone-a-10 is correct": {
			instanceID: "store-gateway-zone-a-10",
			expectedID: 10,
		},
```